### ISSUE TITLE
Feature: Enforce using logical or instead of ternary for falsy assignment

### ISSUE THREAD

**Issue by SimenB** [eslint/eslint#3835]
Really bad title, but I have no idea what to name it.

I'd like the following to error

```js
export default model => model.someValue ? model.someValue : model.someOtherValue;
```

It can be written shorter and (IMO) more readable
```js
export default model => model.someValue || model.someOtherValue;
```

This is mostly stylistic, but it could provide some security against typos, as one of the properties is not typed twice

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by IanVS** [comment]
Sounds like one way would be to add an option on [no-terniary](http://eslint.org/docs/rules/no-ternary) which only prevents ternarys if the first assignment expression is equal to the condition.

**Comment by mysticatea** [comment]
Maybe http://eslint.org/docs/rules/no-unneeded-ternary

**Comment by IanVS** [comment]
Good call, I forgot about that one.

**Comment by SimenB** [comment]
Oh, I didn't know about the second linked. `defaultAssignment: false` does exactly what I want. Thanks!

This is a dupe of #3232
